src/mlib/windows-lean.h

```diff
@@ -0,0 +1,19 @@
+/**
+ * This file is simply a wrapper around <windows.h> and ensures that
+ * WIN32_LEAN_AND_MEAN is defined before including it.
+ */
+#ifndef MLIB_WINDOWS_LEAN_H
+#define MLIB_WINDOWS_LEAN_H
+
+#ifdef __has_include
+#if !__has_include(<windows.h>)
+#error "<mlib/windows-lean.h> is only available when <windows.h> in available."
+#endif
+#endif
+
+#pragma push_macro("WIN32_LEAN_AND_MEAN")
```

Comment:
Per https://docs.microsoft.com/en-us/cpp/preprocessor/push-macro?view=msvc-140, I trust this is portable across various versions going back to VS 2015.